core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/html.kt

```diff
@@ -121,7 +121,7 @@ internal fun AnyFrame.toHtmlData(
     val queue = LinkedList<Pair<AnyFrame, Int>>()
 
     fun AnyFrame.columnToJs(col: AnyCol, rowsLimit: Int): ColumnDataForJs {
-        val values = rows().take(rowsLimit)
+        val values = if (rowsLimit == Int.MAX_VALUE) rows() else rows().take(rowsLimit)
```

Comment:
`take()` already just returns the list if `n >= size`. So this change is not needed.

Reply:
`rows()` returns Iterable which is not a Collection. In this case inside `take()` there is a line 
```kotlin
val list = ArrayList<T>(n)
```
which throws `OutOfMemoryError: Requested array size exceeds VM limit` if n=Int.MAX_VALUE
So this change is necessary to allow using Int.MAX_VALUE as a sentinel value and turn the limit off if we want to render all the data without guessing the upper limit first.

Reply:
I can use `df.count()` to set the limit to the number of rows for an outer dataset to render it fully, but I can't do this for nested dataframes. So I would like to be able to use Int.MAX_VALUE or any other value to turn the limit off and just render everything there is.

Reply:
Ah I see, interesting find! I'd maybe add the check for if rowsLimit < 0, so people can then put in `-1` if they want no limit. I think that's more common.

Reply:
Okay, you convinced me. 
I can only think of one edge case, namely if the list you're taking from is larger than Int.MAX_VALUE, but I think you've reached OOM errors long before that. Unfortunately, there's no "Infinity" option for integers, so this is the best option.

Maybe we'll convert the rows iterable into a `Collection` with `size` to avoid this error in the future.

I will also pass on this case (`Iterable.take()`) to the Kotlin team and see what they think about it.

Reply:
Actually, we could also just change it to `rows().toList().take(rowsLimit)` since it creates a list anyways.

Reply:
Done. I agree that nullable is even more intuitive and hopefully idiomatic  :)